File: src/main/java/org/vladg/jafax/utils/inject/DependencyManager.java
package org.vladg.jafax.utils.inject;

import ast.ASTVisitor;
import ast.repository.NonPersistentRepository;
import ast.repository.model.Class;
import ast.repository.model.File;
import ast.repository.model.Method;
import ast.service.FileService;
import com.google.inject.AbstractModule;
import com.google.inject.Scopes;
import com.google.inject.TypeLiteral;

public class DependencyManager extends AbstractModule {

    @Override
    protected void configure() {
        bind(ASTVisitor.class).in(Scopes.SINGLETON);
        bind(FileService.class).in(Scopes.SINGLETON);
        bind(new TypeLiteral<NonPersistentRepository<File>>() {}).in(Scopes.SINGLETON);
        bind(new TypeLiteral<NonPersistentRepository<Method>>() {}).in(Scopes.SINGLETON);
        bind(new TypeLiteral<NonPersistentRepository<Class>>() {}).in(Scopes.SINGLETON);
    }

}

File: src/main/java/org/vladg/jafax/utils/inject/DependencyManager.kt
package org.vladg.jafax.utils.inject

import com.google.inject.AbstractModule
import com.google.inject.Scopes
import org.vladg.jafax.ast.ASTRequester
import org.vladg.jafax.ast.ASTVisitor
import org.vladg.jafax.ast.unwrapper.*
import org.vladg.jafax.ast.unwrapper.attribute.AttributeUnwrapper
import org.vladg.jafax.ast.unwrapper.clazz.ClassUnwrapper
import org.vladg.jafax.ast.unwrapper.method.MethodInvocationUnwrapper
import org.vladg.jafax.ast.unwrapper.method.MethodUnwrapper

class DependencyManager : AbstractModule() {

    override fun configure() {
        bind(ASTVisitor::class.java).`in`(Scopes.SINGLETON)
        bind(ClassUnwrapper::class.java).`in`(Scopes.SINGLETON)
        bind(MethodUnwrapper::class.java).`in`(Scopes.SINGLETON)
        bind(AttributeUnwrapper::class.java).`in`(Scopes.SINGLETON)
        bind(ContainerService::class.java).`in`(Scopes.SINGLETON)
        bind(MethodInvocationUnwrapper::class.java).`in`(Scopes.SINGLETON)
        bind(ASTRequester::class.java).`in`(Scopes.SINGLETON)
    }

}